x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/LocalExecutionPlanner.java

```diff
@@ -417,12 +419,14 @@ private PhysicalOperation planDissect(DissectExec dissect, LocalExecutionPlanner
         Layout.Builder layoutBuilder = source.layout.builder();
         layoutBuilder.append(dissect.extractedFields());
         final Expression expr = dissect.inputExpression();
-        String[] attributeNames = Expressions.names(dissect.extractedFields()).toArray(new String[0]);
+        // Names in the pattern and layout can differ.
```

Comment:
When is this happening? Can you give an example?

Reply:
Expanded the comment in the latest push.

This happens whenever we call `GeneratingPlan.withGeneratedNames` on `Grok` and `Dissect`.

This enables us to have consistent names in our logical plans, without having to rewrite the format strings for grok and dissect.